### ISSUE TITLE
Miscompilation? of for loop update expression

### ISSUE THREAD

**Issue by sylvarant** [akosba/xjsnark#1]
Compiling a for loop who's update expression is not simply ++ or -- causes a compilation issue as follows:
> for(int j = 0; j < w.length; j = j + 1)
 
Produces a compilation error for the generated java:
> xjsnark\ecKeyKnowledge\EC_KeyKnowledge.java : Syntax error on token ";", delete this token (line: 138)

Due to an unneeded ; 
> for (UnsignedInteger i = new UnsignedInteger(2048, new BigInteger("2")); !((i == UnsignedInteger.instantiateFrom(2048, 0))); 
    i.assign(i.subtract(UnsignedInteger.instantiateFrom(2048, 1)), 2048);)

I encountered this issue as I'm working with for loops for uint_x that do not support ++ or --

**Comment by akosba** [comment]
In the current version,
`       for(int j = 0; j < w.length; j = j + 1) `
or  
`        for(int j = 0; j < w.length; j = j++)`
should compile fine as these are just using the native java types. However, if you use the xjsnark types, like uint_2048 (which the transformed code in your question seems to be related to), we support another loop form in the current version, not the regular for loop shape. Since these types are mostly used for values that are not known during the circuit construction, the current version supports a different bounded while loop, to ensure that a bound always exists. For example, assume this circuit that received some variable input value _n_, and another value _step_;

```
inputs{
   uint_2048 n; 
   uint_2048 step;
}

public static int T = 10000; 
// some bound
```

Later in the code:

```
uint_2048 x = 0;
bwhile( x < n; T  ){
      ...
      x = x + step;
}
```
// Note that T here is a bound on the total number of iterations of this loop. Without specifying T beforehand, it will be difficult or impossible to expect when this part will finish (or a very big bound will have to be assumed). This is why a different syntax is supported for the xjsnark types when they are used with loops.

The above also can written as follows:

```
uint_2048 x = 0;
for(int i = 0; i < T; i++){
      if(x < n){
           ....
           x = x + step;
     }
}
```

So, in summary you can use regular java for loops with native types. If the condition of the loop is input dependent, you can either use a _bwhile_ instead as above, or use an _if_ condition inside the for loop. I have not had a chance to document such use cases yet, but I will try to put an example for how to use the bounded loops in the current version.

Other notes:
- I see in the transformed code " !((i == UnsignedInteger.instantiateFrom(2048, 0)))". This means that you likely used the regular comparison operator with uint_2048. This won't work as desired. Please check the "notes" model code and comments in xjsnark.sandbox  for an illustration of the different operators we use for the xjsnark types. 
- The extra semi-colon was a typo by itself and I removed it.
